main/src/ca/uwaterloo/flix/language/ast/ExecutableAst.scala

```diff
@@ -42,11 +42,12 @@ object ExecutableAst {
                   time: Time,
                   dependenciesOf: Map[Symbol.TableSym, Set[(Constraint, ExecutableAst.Predicate.Body.Positive)]]) extends ExecutableAst
 
-  case class ByteCodes(enumInterfaceByteCodes: Map[EnumSym, (QualName, Array[Byte])],
+  case class ByteCodes(enumInterfaceByteCodes:  Map[EnumSym, (EnumTypeInterfaceName, Array[Byte])],
                        enumClassByteCodes: Map[EnumSym, (Map[(String, Type), Array[Byte]], Map[String, Array[Byte]])],
                        functionalInterfaceByteCodes: Map[Type, (FlixClassName, Array[Byte])],
                        classByteCodes: Map[FlixClassName, Array[Byte]],
-                       tupleByteCode: Map[TupleClassName, Array[Byte]])
+                       tupleByteCode: Map[List[WrappedType], (Array[Byte], Array[Byte])],
+                       ETFusionByteCode: Map[EnumSym, Map[String, Map[List[WrappedType], Array[Byte]]]])
 
```

Comment:
At some point we need to refactor all of this. What is the most general map that can hold all of these?

A Flix type is mapped to a Java type and a Java type has a name (interface or class), so could we do with:

```
Map[JavaName, Array[Byte]]
```
?

Reply:
Sure, can you create a ticket for it?